electrum/address_synchronizer.py

```diff
@@ -970,6 +976,13 @@ def get_utxos(
         if excluded_addresses:
             domain = set(domain) - set(excluded_addresses)
         mempool_height = block_height + 1  # height of next block
+        cache_key = sha256(
+            ','.join(sorted(domain))
+            + f";{mature_only};{confirmed_funding_only};{confirmed_spending_only};{nonlocal_only};{block_height}"
+        )
+        cached = self._get_utxos_cache.get(cache_key)
+        if cached is not None:
+            return cached
```

Comment:
Note that callers might mutate the return value, which would affect the cache and hence subsequent calls. That is an unexpected behaviour change compared to current master.

Instead we should take copies.
```suggestion
            return copy.deepcopy(cached)
```

Reply:
Both applied